packages/gatsby-remark-images/src/index.js

```diff
@@ -103,12 +100,17 @@ module.exports = (
     const fileNameNoExt = fileName.replace(/\.[^/.]+$/, ``)
     const defaultAlt = fileNameNoExt.replace(/[^A-Z0-9]/gi, ` `)
 
-    // TODO
-    // Fade in images on load.
-    // https://www.perpetual-beta.org/weblog/silky-smooth-image-loading.html
-
-    const imageClass = `gatsby-resp-image-image`
-    const imageStyle = `width: 100%; height: 100%; margin: 0; vertical-align: middle; position: absolute; top: 0; left: 0; box-shadow: inset 0px 0px 0px 400px ${
+    const imageStyle = `
+      width: 100%;
+      height: 100%;
+      margin: 0;
+      vertical-align: middle;
+      position: absolute;
+      top: 0;
+      left: 0;
+      transition: opacity 0.5s;
+      transition-delay: 0.5s;
```

Comment:
we can merge those into single `transition: opacity 0.5s 0.5s;`

Reply:
maybe we could move this to that that `classes.js` file (maybe rename it to `constants.js`) where we could have it defined verbose there, but export actually compressed version (with removed all white spaces)? and only dynamically add `box-shadow` property here (as it depends on `backgroundColor`)?

something like:
```js
const imageStyle = `
      width: 100%;
      height: 100%;
      margin: 0;
      vertical-align: middle;
      position: absolute;
      top: 0;
      left: 0;
      transition: opacity 0.5s;
      transition-delay: 0.5s;
`

const compress = css => {
    const ruleExtractor = /(\S+)\s*:\s*(.+);/g

    let rules = ''
    while ((matches = ruleExtractor.exec(imageStyle)) !== null) {
        rules += `${matches[1]}:${matches[2]};`
    }
    return rules
}

exports.compressedImageStyle = compress(css)
```
which should export `width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;`

Reply:
Also just realised - there shouldn't be delay on main image transition I think - it's only useful on placeholder